controllers/runner_controller.go

```diff
@@ -834,6 +834,7 @@ func newRunnerPodWithContainerMode(containerMode string, template corev1.Pod, ru
 		if dockerdContainer != nil {
 			template.Spec.Containers = append(template.Spec.Containers[:dockerdContainerIndex], template.Spec.Containers[dockerdContainerIndex+1:]...)
 		}
+		// TODO: What was the intention here?
 		if runnerContainerIndex < runnerContainerIndex {
 			runnerContainerIndex--
 		}
```

Comment:
@cory-miller Re https://github.com/actions-runner-controller/actions-runner-controller/pull/1794/files#r967413200, it has been there since it's inception in https://github.com/actions-runner-controller/actions-runner-controller/commit/0386c0734cb6fd6ca689ffeb0e76b9ca9cf9512d#diff-f4c239205beda2ef5d9d041809cd93cb32958b6b658fb0b2331c85b853777465R697-R699. It's been unreachable code since then so the first option is to remove this if block entirely.

A better option would be though, to update the code to something like:

```suggestion
		if dockerdContainer != nil {
			template.Spec.Containers = append(template.Spec.Containers[dockerdContainerIndex+1:], template.Spec.Containers[:dockerdContainerIndex]...)
		}
		if dockerdContainerIndex < runnerContainerIndex {
			runnerContainerIndex--
		}
```

I imagine the intention here would be to remove the dockerd sidecar is you're going to use kubernetes container mode(as they are incompatible). If you removed the dockerd sidecar from the containers slice, the runner container's index may change, hence my suggestion.